From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sun, 16 Feb 2025 00:12:56 -0800
Subject: [PATCH] Patch player respawn position chunk load process hang

The base logic attempts to search for a random location within the configured spawn radius

However this also assumes it can create new chunks, which in our case it can't. So this process can loop while the process hangs and the server eventually crashes.

diff --git a/net/minecraft/server/level/ServerPlayer.java b/net/minecraft/server/level/ServerPlayer.java
index 37c0a84758a1526791913898358659de202fa19f..2bfe61a1be93de497bd3feda96a23784b5ef88aa 100644
--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -499,6 +499,8 @@ public class ServerPlayer extends Player implements ca.spottedleaf.moonrise.patc
             int coprime = this.getCoprime(i);
             int randomInt = RandomSource.create().nextInt(i);
 
+            coprime = 0; // Archive - Disable respawn pos adjustment
+
             for (int i1 = 0; i1 < i; i1++) {
                 int i2 = (randomInt + coprime * i1) % i;
                 int i3 = i2 % (max * 2 + 1);
